src/main/java/nextstep/payments/service/PaymentService.java

```diff
@@ -1,10 +1,27 @@
 package nextstep.payments.service;
 
 import nextstep.payments.domain.Payment;
+import nextstep.qna.domain.CoverImage;
+import nextstep.qna.domain.PaidSession;
+import nextstep.qna.domain.Pixel;
+import nextstep.qna.domain.Session;
+import nextstep.qna.domain.session.SessionStatus;
 
 public class PaymentService {
+    public static final Session TEMPORARY_SESSION = PaidSession.of("2023-12-03T00:00:00", "2024-01-03T00:00:00", CoverImage.of("dd", 1024 * 50, "gif", Pixel.of(300), Pixel.of(200)), SessionStatus.RECRUITING, 50, 0, 100000);
+
+
     public Payment payment(String id) {
         // PG사 API를 통해 id에 해당하는 결제 정보를 반환
         return new Payment();
     }
+
+    public void signUpSession(Payment payment) {
+        // 결제 정보를 통해 수강 신청
+//        Session session = SessionRepository.findById(payment.getSessionId());
+        Session session = TEMPORARY_SESSION;
+        if (session.isPossibleToRegister(payment)) {
+            // 수강 신청 성공 후 로직 구현
+        }
```

Comment:
session 에게 상태를 물어보지말고 수강신청을 진행하는것은 어떨까요?

```suggestion
        Payment payment = session.enroll(사용자);
```

그리고 이렇게 리턴값도 하나의 메세지로 볼 수 있을것 같아요!
`session(강의) 에 user(사용자) 를 등록했더니, payment(결제) 결과를 리턴했다.`
이렇게 말로 풀어서 의미가 전달되게요!

메서드는 파라미터 뿐만 아니라 리턴값도 의미를 갖고 있다고 생각하여 저는 이렇게 작성하곤 합니다 :)

Reply:
결제(Payment)에 대한 궁금증인데요, 
이 Payment는 외부에서 결제가 선진행되고, 토큰 등의 인증된 사용자(NsUser)로부터 받았을 요청에 담겨 PaymentService.signUpSession()으로 전달되고 그 Payment 내부의 결제금액을 가지고 판단해서 sessionId와 userId를 업데이트해주는 로직으로 생각해서 짠 코드였는데요,

말씀하신 로직은 Session 내부로 "사용자' 정보만 전달해서 신규 Payment를 도메인 레이어에서 생성하는 듯 보이는데, 이럴 경우 결제가 성공했는지 여부와 금액의 정보는 어디서 확인해야하는걸까요?

Reply:
@AhnHyeonho 

> 유료 강의의 경우 결제는 이미 완료한 것으로 가정하고 이후 과정을 구현한다.
결제를 완료한 결제 정보는 payments 모듈을 통해 관리되며, 결제 정보는 Payment 객체에 담겨 반한된다.

제가 코멘트를 잘못 드렸네요 😅
`payment`는 이미 결제가 완료된 뒤라는 가정을 놓쳤습니다.. 😂

전달드리려했던 부분은 `상태를 물어보지 말고 메세지를 보내시켜라` 였어요!
지금은 서비스에서 `session` 에게 신청할 수 있는 상태를 물어본 뒤 로직을 구현하고 있는데, 이걸 묻지않고 메서드 안에서 처리하라는 말이었습니다 :)

```java
등록결과 = session.enroll(payment);
등록결과_repository.save(등록결과);
```

이런식으로 객체에게 묻지않고 일(?)을 시키면 되지 않을까요? 🤔

등록결과를 영속성으로 관리하지 않는다면,
리턴값 없이 `session.enroll(payment);` 를 호출만 하면 될것 같아요!